src/clojure_lsp/handlers.clj

```diff
@@ -36,6 +36,15 @@
 (def ^:private full-file-range
   (shared/->range {:row 1 :col 1 :end-row 1000000 :end-col 1000000}))
 
+(defmacro process-after-changes [& body]
+  `(let [~'_time (System/nanoTime)]
+     (loop []
+       (if (> (quot (- (System/nanoTime) ~'_time) 1000000) 60000) ; one minute timeout
```

Comment:
I could not repro that even with clojure.core buffer, but I imagine this could happen for crazy long buffers that analysis could take more than one minute, maybe we should use even higher value here?